src/gap.c

```diff
@@ -1523,6 +1523,23 @@ Obj FuncLOAD_DYN (
     if ( info == 0 )
         ErrorQuit( "call to init function failed", 0L, 0L );
 
+    /* info->type used to be a small integer -- test for this case         */
+    if ((UInt)(info->type) < 4)
+        ErrorQuit("LOAD_DYN: kernel module built for older "
+                  "version of GAP",
+                  0L, 0L);
+
+    /* info->type should start with 'D_' */
+    if ((info->type)[0] != 'D' || (info->type)[1] != '_')
```

Comment:
I can never remember what C parses info->type[0] as, so the brackets make sure I get the right one.